ast/parser_test.go

```diff
@@ -4566,6 +4584,141 @@ p { input = "str" }`,
 	}
 }
 
+func TestAnnotationsAugmentedError(t *testing.T) {
+	tests := []struct {
+		note           string
+		module         string
+		expAnnotations []*Annotations
+		expErrorHint   string
+		expErrorRow    int
+	}{
+		{
+			note: "no whitespace after key/value separator",
+			module: "package opa.examples\n" +
+				"\n" +
+				"# METADATA\n" +
+				"# description:p is true\n" +
+				"p := true\n",
+			expErrorHint: "Hint: on line 4, symbol(s) ['p'] immediately following a key/value separator ':' is not a legal yaml space character.",
```

Comment:
Optionally, we could replace the error with the hint text, e.g.
```
1 error occurred: /Users/johan/tmp/non-breaking_whitespace/policy.rego:5: rego_parse_error: symbol(s) ['B'] immediately following a key/value separator ':' is not a legal yaml space character.
```
but there is the concern that we'd mistakenly hide an unrelated error.